master/pkg/schemas/copy_test.go

```diff
@@ -0,0 +1,57 @@
+package schemas
+
+import (
+	"bytes"
+	"encoding/json"
+	"reflect"
+	"testing"
+
+	"gotest.tools/assert"
+)
+
+// Copy is the non-reflect version of copy, but mostly the reflect version is called from other
+// reflect code, so it's defined here in test code.
+func Copy(src interface{}) interface{} {
+	return cpy(reflect.ValueOf(src)).Interface()
+}
+
+func TestCopy(t *testing.T) {
+	var obj []string
+	byts, err := json.Marshal(obj)
+	assert.NilError(t, err)
+	assert.Assert(t, bytes.Equal(byts, []byte("null")))
+
+	src := []string{}
+	byts, err = json.Marshal(src)
+	assert.NilError(t, err)
+	assert.Assert(t, bytes.Equal(byts, []byte("[]")))
+}
+
+func TestCopyAllocatedSlice(t *testing.T) {
```

Comment:
may be nice to have some tests that test nested/filled types (and structs)

Reply:
done